File: SupplierPortal.Web/Modules/ServerTypes/Market/ItemColumns.ts
import { ColumnsBase, fieldsProxy } from "@serenity-is/corelib";
import { Column } from "@serenity-is/sleekgrid";
import { ItemRow } from "./ItemRow";

export interface ItemColumns {
    Id: Column<ItemRow>;
    ItemCode: Column<ItemRow>;
    ItemName: Column<ItemRow>;
}

export class ItemColumns extends ColumnsBase<ItemRow> {
    static readonly columnsKey = 'Market.Item';
    static readonly Fields = fieldsProxy<ItemColumns>();
}

File: SupplierPortal.Web/Modules/Market/Item/ItemColumns.cs
using Serenity.ComponentModel;
using System.ComponentModel;

namespace SupplierPortal.Market.Columns;

[ColumnsScript("Market.Item")]
[BasedOnRow(typeof(ItemRow), CheckNames = true)]
public class ItemColumns
{
    [EditLink, DisplayName("Db.Shared.RecordId"), AlignRight]
    public int Id { get; set; }
    public short ItemCode { get; set; }
    [EditLink]
    public string ItemName { get; set; }
}